File: leetcode/Easy/1.TwoSum_N^2.cpp
#include <bits/stdc++.h>
using namespace std;
#define _GLIBCXX_DEBUG

vector<int> twoSum(vector<int>& nums, int target) {
    const int len = nums.size();
    bool fin = false;
    vector<int> ans;

    for (int i = 0; i < len; i++) {
        for (int j = i + 1; j < len; j++) {
            if (nums[i] + nums[j] == target) {
                ans = {i, j};
                fin = true;
                break;
            }
        }
        if (fin)
            break;
    }
    return ans;
}

int main(void) {
    vector<int> nums = {2, 7, 11, 15};
    vector<int> ans = twoSum(nums, 9);
    cout << ans[0] << " " << ans[1] << endl;
    return 0;
}

// 752ms, 26.68%

File: leetcode/Easy/1.TwoSum_N^2.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	len := len(nums)
	ans := make([]int, 2, 2)
	for i := 0; i < len; i++ {
		for j := i + 1; j < len; j++ {
			if nums[i]+nums[j] == target {
				ans[0] = i
				ans[1] = j
				break
			}
		}
	}
	return ans
}

func main() {
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 18))
}

// 41ms, 27.1%
